adserver/tasks.py

```diff
@@ -1024,6 +1025,47 @@ def update_flight_traffic_fill():
     log.info("Completed updating flight traffic fill")
 
 
+@app.task()
+def daily_flight_hard_stop():
+    """
+    Set flight with a hard stop date to completed.
+
+    This works by setting the sold amount equal to the current fulfilled amount
+    and sending a slack notification about the remaining credit.
+
+    This should be called before `notify_of_completed_flights()`
+    so that the regular wrapup emails can be sent to advertisers.
+    """
+    today = get_ad_day().date()
+
+    for flight in Flight.objects.filter(live=True, hard_stop_date__lte=today):
+        if flight.clicks_remaining() <= 0 and flight.views_remaining() <= 0:
+            continue
+
+        flight.sold_clicks = flight.total_clicks
+        flight.sold_impressions = flight.total_views
```

Comment:
Do we actually want to update the sold data here? Seems like we're losing data on the original buy? Seems like we'd just mark it inactive?